hibernate-core/src/main/java/org/hibernate/query/sqm/tree/expression/AsWrapperSqmExpression.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.query.sqm.tree.expression;
+
+import org.hibernate.query.sqm.SemanticQueryWalker;
+import org.hibernate.query.sqm.SqmExpressible;
+import org.hibernate.query.sqm.tree.SqmCopyContext;
+import org.hibernate.type.BasicType;
+
+public class AsWrapperSqmExpression<T> extends AbstractSqmExpression<T> {
+	private final SqmExpression<?> expression;
+
+	public AsWrapperSqmExpression(SqmExpression<?> expression, Class<T> type) {
+		super(
+				expression.nodeBuilder().getTypeConfiguration().getBasicTypeForJavaType( type ),
+				expression.nodeBuilder()
+		);
+		this.expression = expression;
+	}
+
+	AsWrapperSqmExpression(SqmExpressible<T> type, SqmExpression<?> expression) {
+		super( type, expression.nodeBuilder() );
+		this.expression = expression;
+	}
+
+	@Override
+	public <X> X accept(SemanticQueryWalker<X> walker) {
+		return walker.visitAsWrapperExpression( this );
+	}
+
+	@Override
+	public void appendHqlString(StringBuilder sb) {
+		expression.appendHqlString( sb );
+	}
```

Comment:
We have to decide on a HQL representation for this expression. I propose `wrap(.. as ..)`

```suggestion
		sb.append( "wrap(" );
		expression.appendHqlString( sb );
		sb.append( " as " );
		sb.append( getNodeType().getReturnedClassName() );
		sb.append( ")" );
	}
	
	@Override
	public <X> SqmExpression<X> as(Class<X> type) {
		return expression.as( type );
	}
```